runtime/doc/lua.txt

```diff
@@ -2359,26 +2359,33 @@ find({names}, {opts})                                          *vim.fs.find()*
     The search can be narrowed to find only files or only directories by
     specifying {type} to be "file" or "directory", respectively.
 
+    If {names} is a function, it is called for each file and directory within
+    the traversed directory and it would be passed the base name of each file
+    or directory and the path of the directory being traversed. The function
+    should return `true` if the given file or directory is considered a match.
```

Comment:
Sorry if I wasn't clear, I was suggesting that this should be in the description of the `{names}` parameter below.

My suggestion was a bit prolix, feel free to condense it. Really all I want to see is that someone who's never used this function before can read the docstring and have clarity on what it means for `{names}` to be a function and how that function works (i.e. it takes two arguments, what those arguments are, and what the function should return).

This is also a case where an example is worth a thousand words.